src/charm.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm Jenkins."""
+
+import logging
+from time import sleep
+from typing import TYPE_CHECKING, Any
+
+import requests
+from jenkinsapi.jenkins import Jenkins
+from ops.charm import CharmBase, PebbleReadyEvent
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, Container, MaintenanceStatus
+from ops.pebble import Layer
+
+from path import INITIAL_PASSWORD, JENKINS_HOME, LAST_EXEC, UPDATE_VERSION
+from types_ import Credentials
+
+if TYPE_CHECKING:
```

Comment:
Why this conditional import?

Reply:
This is strictly a type hint only, so importing the type when not in TYPE_CHECKING will throw an error.